File: PR1/PR1_Ej3/Hamburguesa.py


class Hamburguesa:
        def __init__(self):
            self.pan = None
            self.lechuga = None
            self.tomate = None
            self.quesoCabra = None
            self.cebolla = None
            self.pepinillos = None
            self.bacon = None
            self.carne = None

        def __str__(self):

            ingredientes = ''

            if (self.pan != None) :
                ingredientes=ingredientes+f"{self.pan}"
            if (self.lechuga != None) :
                ingredientes=ingredientes+f" {self.lechuga},"
            if (self.tomate != None) :
                ingredientes=ingredientes+f" {self.tomate},"
            if (self.quesoCabra != None) :
                ingredientes=ingredientes+f" {self.quesoCabra},"
            if (self.cebolla != None) :
                ingredientes=ingredientes+f" {self.cebolla},"
            if (self.pepinillos != None) :
                ingredientes=ingredientes+f" {self.pepinillos},"
            if (self.bacon != None) :
                ingredientes=ingredientes+f" {self.bacon},"
            if (self.carne != None) :
                ingredientes=ingredientes+f" {self.carne}."

            return ingredientes

File: PR2/pantalla_pedidos_hamburgueseria/lib/model/Hamburguesa.dart

class Hamburguesa {
  String? pan;
  String? nombre = "";
  String? lechuga;
  String? tomate;
  String? quesoCabra;
  String? cebolla;
  String? pepinillos;
  String? bacon;
  String? carne;
  double? precio = 0;

  Hamburguesa({
    this.pan,
    this.nombre,
    this.lechuga,
    this.tomate,
    this.quesoCabra,
    this.cebolla,
    this.pepinillos,
    this.bacon,
    this.carne,
    this.precio
  });

  double? getPrecio() {
    return this.precio;
  }

  String? getNombre() {
    return this.nombre;
  }

  @override
  String toString() {
    String ingredientes = '';

    if (pan != null) {
      ingredientes += '$pan';
    }
    if (lechuga != null) {
      ingredientes += ' $lechuga,';
    }
    if (tomate != null) {
      ingredientes += ' $tomate,';
    }
    if (quesoCabra != null) {
      ingredientes += ' $quesoCabra,';
    }
    if (cebolla != null) {
      ingredientes += ' $cebolla,';
    }
    if (pepinillos != null) {
      ingredientes += ' $pepinillos,';
    }
    if (bacon != null) {
      ingredientes += ' $bacon,';
    }
    if (carne != null) {
      ingredientes += ' $carne.';
    }

    return ingredientes;
  }

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;

    return other is Hamburguesa &&
        other.nombre == this.nombre &&
        other.pan == this.pan &&
        other.lechuga == this.lechuga &&
        other.tomate == this.tomate &&
        other.quesoCabra == this.quesoCabra &&
        other.cebolla == this.cebolla &&
        other.pepinillos == this.pepinillos &&
        other.bacon == this.bacon &&
        other.carne == this.carne;
  }

  factory Hamburguesa.fromJson(Map<String, dynamic> json) {
    return Hamburguesa(
      pan: json['pan'] as String?,
      nombre: json['nombre'] as String?,
      lechuga: json['lechuga'] as String?,
      tomate: json['tomate'] as String?,
      quesoCabra: json['quesoCabra'] as String?,
      cebolla: json['cebolla'] as String?,
      pepinillos: json['pepinillos'] as String?,
      bacon: json['bacon'] as String?,
      carne: json['carne'] as String?,
      precio: json['precio'] as double?,
    );
  }
}